scipy/signal/tests/test_ltisys.py

```diff
@@ -257,21 +257,59 @@ def test_basic(self):
         assert_allclose(A, [[-2, -3], [1, 0]], rtol=1e-13)
         assert_allclose(B, [[1], [0]], rtol=1e-13)
         assert_allclose(C, [[1, 2]], rtol=1e-13)
-        # N.b. the shape of D returned by tf2ss is (1,), not (1, 1). Sigh.
-        assert_allclose(D, [1], rtol=1e-14)
+        assert_allclose(D, [[1]], rtol=1e-14)
 
         bb, aa = ss2tf(A, B, C, D)
         assert_allclose(bb[0], b, rtol=1e-13)
         assert_allclose(aa, a, rtol=1e-13)
 
     def test_zero_order_round_trip(self):
-        # See Issue #5760
+        # See gh-5760
         tf = (2, 1)
-        ss = tf2ss(*tf)
-        assert_equal(ss, [[0], [0], [0], [2]])
-
-        tf = ss2tf(*ss)
-        assert_equal(tf, [[[2, 0]], [1, 0]])
+        A, B, C, D = tf2ss(*tf)
+        assert_allclose(A, [[0]], rtol=1e-13)
+        assert_allclose(B, [[0]], rtol=1e-13)
+        assert_allclose(C, [[0]], rtol=1e-13)
+        assert_allclose(D, [[2]], rtol=1e-13)
+
+        num, den = ss2tf(A, B, C, D)
+        assert_allclose(num, [[2, 0]], rtol=1e-13)
+        assert_allclose(den, [1, 0], rtol=1e-13)
+
+    def test_simo_round_trip(self):
+        # See gh-5753
+        tf = ([[1, 2], [1, 1]], [1, 2])
+        A, B, C, D = tf2ss(*tf)
+        assert_allclose(A, [[-2]], rtol=1e-13)
+        assert_allclose(B, [[1]], rtol=1e-13)
+        assert_allclose(C, [[0], [-1]], rtol=1e-13)
+        assert_allclose(D, [[1], [1]], rtol=1e-13)
+
+        num, den = ss2tf(A, B, C, D)
+        assert_allclose(num, [[1, 2], [1, 1]], rtol=1e-13)
+        assert_allclose(den, [1, 2], rtol=1e-13)
+
+        tf = ([[1, 0, 1], [1, 1, 1]], [1, 1, 1])
+        A, B, C, D = tf2ss(*tf)
+        assert_allclose(A, [[-1, -1], [1, 0]], rtol=1e-13)
+        assert_allclose(B, [[1], [0]], rtol=1e-13)
+        assert_allclose(C, [[-1, 0], [0, 0]], rtol=1e-13)
+        assert_allclose(D, [[1], [1]], rtol=1e-13)
+
+        num, den = ss2tf(A, B, C, D)
+        assert_allclose(num, [[1, 0, 1], [1, 1, 1]], rtol=1e-13)
+        assert_allclose(den, [1, 1, 1], rtol=1e-13)
+
+        tf = ([[1, 2, 3], [1, 2, 3]], [1, 2, 3, 4])
```

Comment:
Yikes!  Good catch.  Because the code assumes all arrays are 1-D, it can't properly "normalize" `num` when the numerators are misaligned.  I'll need to think about that one for a little bit.